### PR TITLE
Adding Test Framework and Config Tests

### PR DESCRIPTION
**Pull request by grantoesterling**
## What's New
- Adding Jest testing framework
- Adding tests for validating config .jsons
- Adding testing as step in CircleCI
- Converting existing FactoryBot tests to Jest
- Updating README

PR Looks big and scary but mostly whitespace changes
Real stuff is in the two test files, plus slightly changing init function of FactoryBot

## Usage
`yarn test`

Please let me know if there are any more config tests to be added!

### CODE REVIEW

**Comment by grantoesterling** [comment]
apologies for the million commits - tried to get cute with commit squashing but it was being troublesome so i gave up

**Comment by jakerockland** [comment]
> apologies for the million commits - tried to get cute with commit squashing but it was being troublesome so i gave up

No apology needed–we can squash-on-merge!

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,9 +18,22 @@ jobs:
     executor: build-executor
     steps:
       - checkout
-      - run: |
-          npm install
-          timeout 2m npm start || ( [[ $? -eq 124 ]] && echo "WARNING: Timeout reached, but that's OK" )
```
Comment (jakerockland) [inline .circleci/config.yml:23]: @grantoesterling I don't have a strong opinion here–but can you describe why you think we should get rid of this smoke test that just verifies everything runs for a lil?

**Review by jakerockland — APPROVED** [review]

**Comment by jakerockland** [comment]
@Asupkay and @Jbern16 were added for visibility, but they don't need to also approve for this to merge IMHO–LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (grantoesterling) [reply .circleci/config.yml:23]: eep i meant to leave that in actually - good catch. going to add it back

Comment (jakerockland) [reply .circleci/config.yml:23]: Ah rad 👍 SGTM

**Pull request merged**